library/src/main/java/com/novoda/downloadmanager/DefaultNotificationChannelProvider.java

```diff
@@ -0,0 +1,36 @@
+package com.novoda.downloadmanager;
+
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.content.Context;
+import android.os.Build;
+import android.support.annotation.RequiresApi;
+
+class DefaultNotificationChannelProvider implements NotificationChannelProvider {
+
+    private final String channelId;
+    private final String name;
+    private final int importance;
+
+    DefaultNotificationChannelProvider(String channelId, String name, @Importance int importance) {
+        this.channelId = channelId;
+        this.name = name;
+        this.importance = importance;
+    }
+
+    @Override
+    @RequiresApi(Build.VERSION_CODES.O)
+    public void registerNotificationChannel(Context context) {
+        NotificationChannel notificationChannel = new NotificationChannel(channelId, name, importance);
+        notificationManager(context).createNotificationChannel(notificationChannel);
+    }
+
+    @Override
+    public String channelId() {
+        return channelId;
+    }
+
+    private static NotificationManager notificationManager(Context context) {
```

Comment:
this can be inlined

Reply:
I thought it looks better like this. Also because of the `getSystemService` call I need to suppress the nullibility check because `getSystemService` returns Nullable. Should I still inline in that case?

Reply:
Usually I don't like this kind of one-line / one-usage function, I'd be still up to the inlining, but up to you, not a blocker

Reply:
done. Especially since it is a library, I think it makes sense to inline.